calyx-py/calyx/queue_data_gen.py

```diff
@@ -63,29 +65,57 @@ def no_err_cmds_list(queue_size, num_cmds):
     return commands
 
 
-def dump_json(num_cmds, use_rank: bool, no_err: bool, queue_size: Optional[int] = None):
+def dump_json(num_cmds, no_err: bool, queue_size: Optional[int]=None, nwc=False, use_ranks=False, always_true=False):
     """Prints a JSON representation of the data to stdout.
     The data itself is populated randomly, following certain rules:
     - It has three "memories": `commands`, `values`, and `ans_mem`.
-    - The `commands` memory has `num_cmds` items, which are 0, 1, or 2.
-      0: pop, 1: peek, 2: push
-      If the `no_err` flag is set, then items are chosen from 0 and 2 using a helper.
+    - Optional memories `ranks` and `times` are included for queues primed for non-work-conserving algorithms.
+    - The `commands` memory has `num_cmds` items, which range from 0-2 for work-conserving policies,
+        and from 0-4 for non-work-conserving. They are as follows:
+
+    FOR WORK-CONSERVING POLICIES
+        0 : pop
+        1 : peek
+        2 : push
+    
+    FOR NON-WORK-CONSERVING POLICIES
+        0 : pop by predicate
+        1 : peek by predicate
+        2 : push
+        3 : pop by value
+        4 : peek by value
+
+    If the `no_err` flag is set and the policy of work-conserving,
+    then items are chosen from 0 and 2 using a helper.
+
+    If the `always-true` flag is set to True for non-work-conserving policies,
```

Comment:
As I see it, it's just there for the totality of the function. There theoretically isn't anything to stop the user from passing in `True` for both argments `nwc` and `always-true`. It's definitely silly but I don't think it's harmful, so I think it's fine to keep it like this and assume anyone who uses this function (which will most likely just be us) would make the more logical decision.

I agree about the other point though!

Reply:
After giving it some thought, it strikes me that we could just remove the `always-true` parameter altogether. Since by the previous comment, `nwc` should determine whether the predicate is always true or not. What do you think?

Reply:
Yes, ☝️  was sorta my angle earlier! Could just nix it. Keep it if you see a use-case, but otherwise I propose we remove it for now and add it back if we learn of a use-case.

Reply:
Sounds good!

Reply:
Done now!